app/controllers/api/v1/targets_controller.rb

```diff
@@ -8,6 +8,8 @@ def index
       def create
         authorize Target
         @target = TargetService.new(current_user, resource_params).create!
+      rescue MaxTargetsAmountReachedError
```

Comment:
I understand your point, and I like it. In first place, I did it in the policy because I think the limit of the number of targets that a user can create is a characteristic of the user and not of the targets. Anyway, I have a question: what happens if the limit of the targets creation is not in all users? For example, an administrator or a VIP user may not have limits on the number of targets they can create. All the validations in the model always occur, or can they depend on the context of the moment, validating only some things? For example, can the method in the model have a condition that if the user if a VIP or admin, they return true by default? Is this ok?

Reply:
On that case, you can write that logic on the model validation method, same as you would do on the policy.

If your point is how to handle complex validations and that can be too much logic on the model, there is an approach like validators, a class that have all the model validation on place, so you have the models clean.
For example:

```
class TargetValidator

  def initialize(model)
    @model = model
  end
  def valid?
    # your custom and complex validation logic here
  end
end
```